tests/publiccloud/cloud_netconfig.pm

```diff
@@ -12,77 +12,119 @@
-    }
-    elsif (is_gce()) {
-        $query_meta_ipv4_cmd =
-          'curl -H "Metadata-Flavor: Google" '
-          . "\"http://$pc_meta_api_ip/computeMetadata/v1/instance/"
-          . 'network-interfaces/0/ip"';
-    } else {
-        die("Unsupported public cloud provider.");
+    if (is_azure) {
+        record_info('eth1', 'Get public IP address for eth1');
+        my $local_eth1_ip = $instance->ssh_script_output(qq(ip -4 -o a s eth1 primary | grep -Po "inet \\K[\\d.]+" | head -n1));
+        chomp($local_eth1_ip);
+        my $metadata_eth1_ip = $provider->query_metadata($instance, ifNum => '1', addrCount => '0');
+        die("Failed to get interface 1 IP from metadata server") unless length($metadata_eth1_ip);
+        assert_equals($metadata_eth1_ip, $local_eth1_ip, 'Locally assigned eth1 IP does not equal the IP retrieved from CSP metadata service.');
+
+        record_info('secondary', 'Make sure each interface has also secondary IP address');
+        my $local_eth0_secondary_ip = $instance->ssh_script_output(qq(ip -4 -o a s dev eth0 secondary | grep -Po "inet \\K[\\d.]+" | head -n1));
+        chomp($local_eth0_secondary_ip);
+        my $local_eth1_secondary_ip = $instance->ssh_script_output(qq(ip -4 -o a s dev eth1 secondary | grep -Po "inet \\K[\\d.]+" | head -n1));
+        chomp($local_eth1_secondary_ip);
+
+        record_info('Routes', 'Check that there is default route for each interface');
+        die('There is no IP rule') if ($instance->ssh_script_output('ip route show default dev eth0 | wc -l') == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output('ip route show default dev eth1 | wc -l') == 0);
+
+        record_info('Connections', 'Make HTTPS connection from each interface and check they have different public IP address');
+        $instance->ssh_assert_script_run('curl -s https://wtfismyip.com/text');
+        my $eth0_public_ip = $instance->ssh_script_output('curl --fail --interface eth0 -s https://wtfismyip.com/text');
+        my $eth1_public_ip = $instance->ssh_script_output('curl --fail --interface eth1 -s https://wtfismyip.com/text');
+        assert_not_equals($eth0_public_ip, $eth1_public_ip, "The connection from eth0 shouldn't have the same public IPv4 address as connection from eth1");
+
+        record_info('Rules', 'Make sure there are IP rules for each IP address');
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_secondary_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_secondary_ip | wc -l") == 0);
+
+        record_info('Neighbors', 'Make sure there is at least one ARP neighbor on each interface');
```

Comment:
Better as code comment?

Also, nice idea for making sure there is some activity :+1:

Reply:
You're right that comments are more visible in the code but `record_info` makes nice openQA square so it's easier to review in which check the test failed. WDYT?